### PR TITLE
DarkMode: fix three bugs in components

### PR DESCRIPTION
**Pull request by REJack**
I've fixed three dark mode bugs.

- Hr: reflect daisyUI's divider style 
- Nav: update border-color to work good light and dark 
- ThemeToggle: fix initial theme setter from 3 possible sources
  also fixed in this commit:
  - replaced setAttribute on class to classList add/remove to avoid html class reset/override
  - fix usage of theme-controller to avoid wrong switching
  - add missing focus outline if ThemeToggle has class `btn` set

Let me know if you want some screenshots to compare the before and after.

### CODE REVIEW

**Comment by robsontenorio** [comment]
Please, attach some screenshots.

File: src/View/Components/ThemeToggle.php [hunk 1]
```diff
@@ -27,32 +27,50 @@ public function render(): View|Closure|string
...
+                    window.localStorage.setItem("mary-theme", JSON.stringify(endTheme))
+                    
+                    document.documentElement.setAttribute("data-theme", endTheme)
+                    if (endTheme === "dark") {
+                        document.documentElement.classList.add("dark")
+                        document.documentElement.classList.remove("light")
+                    } else {
+                        document.documentElement.classList.add("light")
+                        document.documentElement.classList.remove("dark")
+                    }
```
Comment (robsontenorio) [inline src/View/Components/ThemeToggle.php:73]: As you navigate through routes it breaks

image

File: src/View/Components/Hr.php [hunk 2]
```diff
@@ -28,10 +28,10 @@ public function progressTarget(): ?string
     public function render(): View|Closure|string
     {
         return <<<'HTML'
-                <div class="h-[2px] border border-t-base-100 my-5">
+                <div class="h-[2px] border border-base-content/10 my-5">
                     <progress
-                        class="progress progress-primary hidden h-[1px]"
-                        wire:loading.class="!h-[2px] !block"
+                        class="progress progress-primary hidden -mt-[1px] h-[2px]"
+                        wire:loading.class="!block"
```
Comment (robsontenorio) [inline src/View/Components/Hr.php:34]: On light theme this makes the line larger. It should looks like what was before.

File: src/View/Components/ThemeToggle.php (hunk 1 above)
Comment (REJack) [reply src/View/Components/ThemeToggle.php:77]: Bug fixed, but these errors show's that your code also got executed on every page switch.

File: src/View/Components/Hr.php (hunk 2 above)
Comment (REJack) [reply src/View/Components/Hr.php:34]: fixed, both now thinner again

**Comment by REJack** [comment]
> Please, attach some screenshots.

image
image

**Comment by robsontenorio** [comment]
You have fixed the console error. But there is another bug:

- After navigate through some routes the toggle does not switch the theme.

**Comment by REJack** [comment]
👍🏻 I'll investigate that.

**Comment by robsontenorio** [comment]
Thank you! Solved by #364

**Pull request closed without merging**